### ISSUE TITLE
Custom readers for dask_image using pims

### ISSUE THREAD

**Issue by astrophysaxist** [dask/dask-image#230]
Hi, I wanted to raise the request to have a way to incorporate custom readers using `pims` as described [here](http://soft-matter.github.io/pims/v0.5/custom_readers.html).

My specific use case is I often work with FITS format files. It's pretty straightforward to write a custom reader that works with `pims.open()`, but it seems that currently with `dask_images.imread.imread()`, the `pims.open()` call does not find a new reader. This may just be a scope issue; I've been playing around with this in a Jupyter notebook so far.

**Comment by jakirkham** [comment]
The thing to check would be whether Dask workers in the deployment are able to open using the relevant PIMS plugin. If that's not working, that's where one would want to start debugging

**Comment by astrophysaxist** [comment]
Hi @jakirkham , do you mean try setting up a dask cluster and try opening images with `pims.open`? Or are you suggesting something else?

**Comment by jakirkham** [comment]
Yeah exactly

**Comment by GenevieveBuckley** [comment]
It's worth noting that there is a separate `imread` function in dask itself (not dask-image), which allows you to pass in specific reader functions. You might find that useful: https://docs.dask.org/en/latest/array-api.html?highlight=image#dask.array.image.imread

**Comment by astrophysaxist** [comment]
Thanks @jakirkham and @GenevieveBuckley both your suggestions were helpful for me to really narrow down what's going on.

In running the `pims.open` on dask workers I uncovered that there seems to be an issue with the plugin from `imageio` correctly identifying `.fits` formats in the absence of a `format=` keyword. See the issue [here](https://github.com/imageio/imageio/issues/636#issue-896253345) I submitted if you're interested.

As for the `dask.array.image.imread` workaround, it can accomplish what I need. When the `imageio` plugin issue fix gets merged it should make this issue obsolete and `dask_image` should be able to read FITS format just fine via `imageio`.
